trino-aws-proxy/src/test/java/io/trino/aws/proxy/server/testing/TestingCredentialsRolesProvider.java

```diff
@@ -69,7 +69,9 @@ public Optional<Credentials> credentials(String emulatedAccessKey, Optional<Stri
                 checkState(emulatedAccessKey.equals(session.sessionCredential.accessKey()), "emulatedAccessKey and session accessKey mismatch");
 
                 Credentials originalCredentials = requireNonNull(credentials.get(session.originalEmulatedAccessKey), "original credentials missing for: " + session.originalEmulatedAccessKey);
-                return Optional.of(Credentials.build(session.sessionCredential, originalCredentials.remote()));
+                return Optional.of(originalCredentials.remoteSessionRole()
+                        .map(remoteSessionRole -> Credentials.build(session.sessionCredential, originalCredentials.requiredRemoteCredential(), remoteSessionRole))
```

Comment:
Note that without this, `TestProxiedEmulatedAndRemoteAssumedRoleRequests` did not work.

Using a session token would result in us not using a remote role, so the test was actually not testing anything I don't think